File: android/src/test/java/com/liferay/mobile/android/DLAppServiceTest.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android;

import com.liferay.mobile.android.exception.ServerException;
import com.liferay.mobile.android.service.BatchSessionImpl;
import com.liferay.mobile.android.v62.dlapp.DLAppService;

import java.io.IOException;

import java.nio.charset.StandardCharsets;

import org.json.JSONArray;
import org.json.JSONObject;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * @author Bruno Farache
 */
public class DLAppServiceTest extends BaseTest {

	public static final String FILE_ENTRY_ID = "fileEntryId";

	public static final String MIME_TYPE = "text/plain";

	public static final int PARENT_FOLDER_ID = 0;

	public static final String SOURCE_FILE_NAME = "test.properties";

	public static final String TITLE = "title";

	public DLAppServiceTest() throws IOException {
		super();
	}

	public JSONObject addFileEntry() throws Exception {
		DLAppService service = new DLAppService(session);
		long repositoryId = props.getGroupId();

		byte[] bytes = "Hello".getBytes(StandardCharsets.UTF_8);

		return service.addFileEntry(
			repositoryId, PARENT_FOLDER_ID, SOURCE_FILE_NAME, MIME_TYPE,
			SOURCE_FILE_NAME, "", "", bytes, null);
	}

	@Test
	public void addFileEntryBytes() throws Exception {
		JSONObject jsonObj = addFileEntry();

		assertEquals(SOURCE_FILE_NAME, jsonObj.get(TITLE));

		deleteFileEntry(jsonObj.getLong(FILE_ENTRY_ID));
	}

	@Test
	public void addFolder() throws Exception {
		DLAppService service = new DLAppService(session);
		long repositoryId = props.getGroupId();

		JSONObject jsonObj = service.addFolder(
			repositoryId, PARENT_FOLDER_ID, _FOLDER_NAME, "", null);

		assertEquals(_FOLDER_NAME, jsonObj.get(_NAME));

		int count = service.getFoldersCount(
			repositoryId, PARENT_FOLDER_ID, 0, false);

		assertEquals(1, count);

		deleteFolder();
	}

	@Test
	public void addFoldersBatch() throws Exception {
		BatchSessionImpl batch = new BatchSessionImpl(session);

		DLAppService service = new DLAppService(batch);
		long repositoryId = props.getGroupId();

		service.addFolder(
			repositoryId, PARENT_FOLDER_ID, _FOLDER_NAME, "", null);

		service.addFolder(
			repositoryId, PARENT_FOLDER_ID, _FOLDER_NAME_2, "", null);

		JSONArray jsonArray = batch.invoke();

		assertEquals(_FOLDER_NAME, jsonArray.getJSONObject(0).get(_NAME));
		assertEquals(_FOLDER_NAME_2, jsonArray.getJSONObject(1).get(_NAME));

		deleteFoldersBatch(batch);
	}

	public void deleteFileEntry(long fileEntryId) throws Exception {
		DLAppService service = new DLAppService(session);
		service.deleteFileEntry(fileEntryId);
	}

	public void deleteFolder() throws Exception {
		DLAppService service = new DLAppService(session);
		long repositoryId = props.getGroupId();

		service.deleteFolder(repositoryId, PARENT_FOLDER_ID, _FOLDER_NAME);

		try {
			service.getFolder(repositoryId, PARENT_FOLDER_ID, _FOLDER_NAME);

			fail();
		}
		catch (ServerException se) {
			String message = se.getMessage();

			assertTrue(message.startsWith("No DLFolder exists with the key"));
		}
	}

	public void deleteFoldersBatch(BatchSessionImpl batch) throws Exception {
		DLAppService service = new DLAppService(batch);
		long repositoryId = props.getGroupId();

		service.deleteFolder(repositoryId, PARENT_FOLDER_ID, _FOLDER_NAME);
		service.deleteFolder(repositoryId, PARENT_FOLDER_ID, _FOLDER_NAME_2);

		JSONArray jsonArray = batch.invoke();

		assertEquals(2, jsonArray.length());
	}

	private static final String _FOLDER_NAME = "test";

	private static final String _FOLDER_NAME_2 = "test2";

	private static final String _NAME = "name";

}

File: windows/Liferay.SDK.Test/DLAppServiceTest.cs
//------------------------------------------------------------------------------
// <copyright file="DLAppServiceTest.cs">
//    Copyright (c) 2014-present Andrea Di Giorgi
//
//    Permission is hereby granted, free of charge, to any person obtaining a
//    copy of this software and associated documentation files (the "Software"),
//    to deal in the Software without restriction, including without limitation
//    the rights to use, copy, modify, merge, publish, distribute, sublicense,
//    and/or sell copies of the Software, and to permit persons to whom the
//    Software is furnished to do so, subject to the following conditions:
//
//    The above copyright notice and this permission notice shall be included in
//    all copies or substantial portions of the Software.
//
//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
//    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
//    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
//    DEALINGS IN THE SOFTWARE.
// </copyright>
// <author>Andrea Di Giorgi</author>
// <website>https://github.com/Ithildir/liferay-sdk-builder-windows</website>
//------------------------------------------------------------------------------

using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Liferay.SDK.Exceptions;
using Liferay.SDK.Service.V62.DLApp;
using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;

namespace Liferay.SDK.Test
{
	[TestClass]
	public class DLAppServiceTest : TestBase
	{
		private const string FolderName = "test";

		private const string FolderName2 = "test2";

		private const string MimeType = "text/plain";

		private const long ParentFolderId = 0;

		private const string SourceFileName = "test.properties";

		[TestMethod]
		public async Task TestAddFileEntryBytes()
		{
			var service = new DLAppService(this.Session);
			var repositoryId = GroupId;

			var bytes = Encoding.UTF8.GetBytes("Hello");

			var fileEntry = await service.AddFileEntryAsync(repositoryId, ParentFolderId, SourceFileName, MimeType, SourceFileName, string.Empty, string.Empty, bytes, null);

			Assert.AreEqual(SourceFileName, fileEntry.title);

			await service.DeleteFileEntryAsync(fileEntry.fileEntryId);
		}

		[TestMethod]
		public async Task TestAddFileEntryStream()
		{
			var service = new DLAppService(this.Session);
			var repositoryId = GroupId;

			var bytes = Encoding.UTF8.GetBytes("Hello");
			var stream = new MemoryStream(bytes);

			var fileEntry = await service.AddFileEntryAsync(repositoryId, ParentFolderId, SourceFileName, MimeType, SourceFileName, string.Empty, string.Empty, stream, null);

			Assert.AreEqual(SourceFileName, fileEntry.title);

			await service.DeleteFileEntryAsync(fileEntry.fileEntryId);
		}

		[TestMethod]
		public async Task TestAddFolder()
		{
			var service = new DLAppService(this.Session);
			var repositoryId = GroupId;

			var folder = await service.AddFolderAsync(repositoryId, ParentFolderId, FolderName, string.Empty, null);

			Assert.AreEqual(FolderName, folder.name);

			var count = await service.GetFoldersCountAsync(repositoryId, ParentFolderId, 0, false);

			Assert.AreEqual(1, count);

			await this.DeleteFolder();
		}

		[TestMethod]
		public async Task TestAddFoldersBatch()
		{
			var batch = new BatchSession(this.Session);

			var service = new DLAppService(batch);
			var repositoryId = GroupId;

			await service.AddFolderAsync(repositoryId, ParentFolderId, FolderName, string.Empty, null);

			await service.AddFolderAsync(repositoryId, ParentFolderId, FolderName2, string.Empty, null);

			var array = await batch.InvokeAsync();

			Assert.AreEqual(FolderName, array.ElementAt(0).name);
			Assert.AreEqual(FolderName2, array.ElementAt(1).name);

			await this.DeleteFoldersBatch(batch);
		}

		private async Task DeleteFolder()
		{
			var service = new DLAppService(this.Session);
			var repositoryId = GroupId;

			await service.DeleteFolderAsync(repositoryId, ParentFolderId, FolderName);

			try
			{
				await service.GetFolderAsync(repositoryId, ParentFolderId, FolderName);

				Assert.Fail();
			}
			catch (ServerException se)
			{
				Assert.IsTrue(se.Message.StartsWith("No DLFolder exists with the key"));
			}
		}

		private async Task DeleteFoldersBatch(BatchSession batch)
		{
			var service = new DLAppService(batch);
			var repositoryId = GroupId;

			await service.DeleteFolderAsync(repositoryId, ParentFolderId, FolderName);
			await service.DeleteFolderAsync(repositoryId, ParentFolderId, FolderName2);

			var array = await batch.InvokeAsync();

			Assert.AreEqual(2, array.Count());
		}
	}
}